sphinx_immaterial/apidoc_formatting.py

```diff
@@ -225,92 +111,80 @@ def depart_caption(self, node: docutils.nodes.Element) -> None:
-        if sphinx.version_info >= (4, 3):
-            return prefix
-        parts = prefix.strip().split(" ")
-        if "property" in parts:
-            parts.remove("property")
-        if parts:
-            return " ".join(parts) + " "
-        return ""
-
-    directive_cls.get_signature_prefix = get_signature_prefix
-
-
-def _monkey_patch_pyattribute_handle_signature(
-    directive_cls: Type[sphinx.domains.python.PyObject],
-):
-    """Modifies PyAttribute or PyVariable to improve styling of signature."""
-
-    def handle_signature(
-        self, sig: str, signode: sphinx.addnodes.desc_signature
-    ) -> Tuple[str, str]:
-        result = super(directive_cls, self).handle_signature(sig, signode)
-        typ = self.options.get("type")
-        if typ:
-            signode += sphinx.addnodes.desc_sig_punctuation("", " : ")
-            signode += sphinx.domains.python._parse_annotation(typ, self.env)
-
-        value = self.options.get("value")
-        if value:
-            signode += sphinx.addnodes.desc_sig_punctuation("", " = ")
-            signode += docutils.nodes.literal(
-                text=value, classes=["code", "python"], language="python"
-            )
-        return result
-
-    directive_cls.handle_signature = handle_signature
+    def depart_desc_inline(self, node: sphinx.addnodes.desc_inline) -> None:
+        self.body.append("</code>")
 
 
-def _monkey_patch_parameterlist_to_support_subscript():
-    desc_parameterlist = sphinx.addnodes.desc_parameterlist
+def _wrap_signature(node: sphinx.addnodes.desc_signature, limit: int):
+    """Wraps long function signatures.
 
-    def astext(self: desc_parameterlist) -> str:
-        open_paren, close_paren = self.get("parens", ("(", ")"))
-        return f"{open_paren}{super(desc_parameterlist, self).astext()}{close_paren}"
+    Adds the `sig-wrap` class which causes each parameter to be displayed on a
+    separate line.
+    """
+    node_text = node.astext()
+    if len(node_text) > limit:
+        node["classes"].append("sig-wrap")
 
-    desc_parameterlist.astext = astext
 
+def _wrap_signatures(
+    app: sphinx.application.Sphinx,
+    domain: str,
+    objtype: str,
+    content: docutils.nodes.Element,
+) -> None:
+    if not app.config.html_wrap_signatures_with_css(domain):
+        return
+    signatures = content.parent[:-1]
+    for signature in signatures:
+        _wrap_signature(
+            signature, app.config.html_wrap_signatures_with_css_column_limit
+        )
 
-def _monkey_patch_napoleon_admonition_classes():
-    GoogleDocstring = sphinx.ext.napoleon.docstring.GoogleDocstring
 
-    def _add_admonition_class(method_name: str, class_name: str) -> None:
-        orig_method = getattr(GoogleDocstring, method_name)
+def _normalize_wrap_signatures_predicate(value):
+    if value is None:
+        value = True
+    if isinstance(value, bool):
+        return lambda _domain: value
+    if callable(value):
```

Comment:
What's the use case for accepting a callable. Is it meant to be a dynamic filter?

Reply:
Removed.  I originally documented that supported a dynamic filter, but then removed that to simplify things.